front50-gcs/src/main/java/com/netflix/spinnaker/front50/model/GcsStorageService.java

```diff
@@ -499,15 +510,24 @@ private void purgeOldVersions(String path) throws Exception {
 
   @Override
   public long getLastModified(ObjectType objectType) {
+      return getLastModifiedHelper(objectType, true);
+  }
+
+  private long getLastModifiedHelper(ObjectType objectType, boolean expected) {
       String path = daoRoot(objectType.group) + '/' + LAST_MODIFIED_FILENAME;
       try {
           return obj_api.get(bucketName, path).execute().getUpdated().getValue();
       } catch (HttpResponseException e) {
           if (e.getStatusCode() == 404) {
-              log.error("No timestamp file at {}. Creating a new one.", path);
-              writeLastModified(objectType.group);
+              if (expected) {
+                  log.error("No timestamp file at {}. Creating a new one.", path);
```

Comment:
Not sure I'd consider a missing last modified file to be an error ... but don't know the whole backstory that's motivating these changes.